src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/CropEdgeGeometry.cpp

```diff
@@ -0,0 +1,410 @@
+#include "CropEdgeGeometry.hpp"
+
+#include "simplnx/DataStructure/DataArray.hpp"
+#include "simplnx/DataStructure/DataGroup.hpp"
+#include "simplnx/DataStructure/Geometry/EdgeGeom.hpp"
+#include "simplnx/Utilities/DataArrayUtilities.hpp"
+#include "simplnx/Utilities/ParallelDataAlgorithm.hpp"
+
+using namespace nx::core;
+
+namespace
+{
+/**
+ * @brief
+ * @tparam T
+ */
+template <typename T>
+class CropEdgeGeomArray
+{
+public:
+  CropEdgeGeomArray(const IDataArray& oldCellArray, IDataArray& newCellArray, const AttributeMatrix& srcAttrMatrix, const std::vector<bool>& tupleMask, const std::atomic_bool& shouldCancel)
+  : m_OldCellStore(oldCellArray.template getIDataStoreRefAs<AbstractDataStore<T>>())
+  , m_NewCellStore(newCellArray.template getIDataStoreRefAs<AbstractDataStore<T>>())
+  , m_SrcAttrMatrix(srcAttrMatrix)
+  , m_TupleMask(tupleMask)
+  , m_ShouldCancel(shouldCancel)
+  {
+  }
+
+  ~CropEdgeGeomArray() = default;
+
+  CropEdgeGeomArray(const CropEdgeGeomArray&) = default;
+  CropEdgeGeomArray(CropEdgeGeomArray&&) noexcept = default;
+  CropEdgeGeomArray& operator=(const CropEdgeGeomArray&) = delete;
+  CropEdgeGeomArray& operator=(CropEdgeGeomArray&&) noexcept = delete;
+
+  void operator()() const
+  {
+    convert();
+  }
+
+protected:
+  void convert() const
+  {
+    usize newIndex = 0;
+    for(usize i = 0; i < m_SrcAttrMatrix.getNumTuples(); ++i)
+    {
+      if(m_TupleMask[i])
+      {
+        CopyFromArray::CopyData(m_OldCellStore, m_NewCellStore, newIndex, i, 1);
+        newIndex++;
+      }
+    }
+  }
```

Comment:
```suggestion
  void operator()() const
  {
    usize newIndex = 0;
    for(usize i = 0; i < m_SrcAttrMatrix.getNumTuples(); ++i)
    {
      if(m_TupleMask[i])
      {
        CopyFromArray::CopyData(m_OldCellStore, m_NewCellStore, newIndex, i, 1);
        newIndex++;
      }
    }
  }
```
I don't think you need a wrapper here; it just bloats generated class obj.

Unless there is a child class somewhere that needs a unique overload